app_config_b2c.py

```diff
@@ -0,0 +1,33 @@
+import os
+
+b2c_tenant = "fabrikamb2c"
+signupsignin_user_flow = "b2c_1_signupsignin1"
```

Comment:
Is there a reason you're using a different policy? I think most of our samples use these:
"b2c_1_susi", "b2c_1_edit_profile", "b2c_1_reset"
Do you have access to the fabrikam b2c tenant? Asking because someone could enable MFA on your su_si policy or something, so it's nice if we are all using the same ones across the samples.

Reply:
what about the password reset? i'm not sure when it will be moved into the su_si in flow, but right now, devs have to catch the error message coming back from B2C AAD and then run the password reset user flow. At the moment, definitely something to think about adding...@jmprieur thoughts?

---
In reply to: [341422121](https://github.com/Azure-Samples/ms-identity-python-webapp/pull/4#discussion_r341422121) [](ancestors = 341422121)

Reply:
yes, I think that we should catch the error for now, @jennyf19 
@rayluo @abhidnya13 : the .NET core is:
https://github.com/Azure-Samples/active-directory-b2c-dotnet-desktop/blob/365e548fe2ef9954188b9332d62c82ce3e56904d/active-directory-b2c-wpf/MainWindow.xaml.cs#L41-L62

```CSharp
            try
            {
                ResultText.Text = "";
                authResult = await (app as PublicClientApplication).AcquireTokenInteractive(App.ApiScopes)
                    .WithParentActivityOrWindow(new WindowInteropHelper(this).Handle)
                    .ExecuteAsync();


                DisplayUserInfo(authResult);
                UpdateSignInState(true);
            }
            catch (MsalException ex)
            {
                try
                {
                    if (ex.Message.Contains("AADB2C90118"))
                    {
                        authResult = await (app as PublicClientApplication).AcquireTokenInteractive(App.ApiScopes)
                            .WithParentActivityOrWindow(new WindowInteropHelper(this).Handle)
                            .WithPrompt(Prompt.SelectAccount)
                            .WithB2CAuthority(App.AuthorityResetPassword)
                            .ExecuteAsync();
                    }
```

Reply:
> Is there a reason you're using a different policy? I think most of our samples use these:
> "b2c_1_susi", "b2c_1_edit_profile", "b2c_1_reset"
> Do you have access to the fabrikam b2c tenant? Asking because someone could enable MFA on your su_si policy or something, so it's nice if we are all using the same ones across the samples.

This is probably something we should discuss out-of-band. The B2C tutorials all use the format that @rayluo has included here, and are referenced as such throughout the rest of the doc set. Upon completion of the userflow creation tutorial, the reader has the following policies in place:

![image](https://user-images.githubusercontent.com/13394541/68041303-8563a880-fc8d-11e9-982e-de334f4760c4.png)

I think there's a larger "who's authoritative, here" discussion to have, but my vote would be to stick with what Ray has in the PR. The longer versions are more explicit, as not everyone might know at a glance what "su_si" means, or that "reset" is resetting the _password_.

Reply:
@mmacy i had no idea...we really need to clean up our .NET samples then. I don't even see those flows in our fabrikam b2c tenant. so who has access to those user flows? asking because when we've presented to customers, we like to show how things are set up. I'm in the AzureAD B2C Demo directory.

Reply:
@jennyf19 Good question. I don't know who owns, manages, or has admin perms to the fabrikamb2c tenant. Finding that out is probably our step 0 to getting docs, samples, and demos in sync. Something to put on the agenda for next sync?

Reply:
@mmacy Thanks. yes, let's talk about this. i have access to `fabrikamb2c.onmicrosoft.com` tenant, which is where all the other samples are pulling their user flows and apis from, as well as the apps. everything is registered there.

Reply:
Other than all the informative discussion above, do we have any actionable items here for this PR?

Currently there are 2 occurrences of `b2c_1_signupsignin1` in this PR.

* The `b2c_1_signupsignin1` in `app_config_b2c.py` is in fact a meaningless placeholder. It could just be a string "b2c_1_XYZ" and this sample and README_B2C.md would still be considered correct, because our `README_B2C.md` is already guiding reader to replace all the relevant values. We don't expect reader to use our template as-is, it won't work anyway, because we won't publish our client_secret.

* The `b2c_1_signupsignin1` is also currently mentioned in `README_B2C.md` like this:

> Replace the value of `signupsignin_user_flow` with `b2c_1_signupsignin1`

Using that verbatim value is assuming the reader have followed [the prerequisite](https://docs.microsoft.com/en-us/azure/active-directory-b2c/tutorial-create-user-flows). But one may argue that we should use a more generic tone, such as:

> Replace the value of `signupsignin_user_flow` with *the sign up and sign in user flow name that you chose when you create that user flow*.

Regardless of what naming convention we would agree with, these values are all ultimately decided by what the customer actually set up after they follow [the prerequisite](https://docs.microsoft.com/en-us/azure/active-directory-b2c/tutorial-create-user-flows).

Reply:
> yes, I think that we should catch the error for now, @jennyf19
> @rayluo @abhidnya13 : the .NET core is:
> https://github.com/Azure-Samples/active-directory-b2c-dotnet-desktop/blob/365e548fe2ef9954188b9332d62c82ce3e56904d/active-directory-b2c-wpf/MainWindow.xaml.cs#L41-L62
> 
> ```cs
>             try
>             {
>                 ResultText.Text = "";
>                 authResult = await (app as PublicClientApplication).AcquireTokenInteractive(App.ApiScopes)
>                     .WithParentActivityOrWindow(new WindowInteropHelper(this).Handle)
>                     .ExecuteAsync();
> 
> 
>                 DisplayUserInfo(authResult);
>                 UpdateSignInState(true);
>             }
>             catch (MsalException ex)
>             {
>                 try
>                 {
>                     if (ex.Message.Contains("AADB2C90118"))
>                     {
>                         authResult = await (app as PublicClientApplication).AcquireTokenInteractive(App.ApiScopes)
>                             .WithParentActivityOrWindow(new WindowInteropHelper(this).Handle)
>                             .WithPrompt(Prompt.SelectAccount)
>                             .WithB2CAuthority(App.AuthorityResetPassword)
>                             .ExecuteAsync();
>                     }
> ```

Was not aware that Reset Password would need to be handled differently. Did not encounter such guidance in the docs I read. :-(
There is not enough time to adjust this one today. Let's treat it as a missing feature of this PR for now. We will make some change later, either in this PR (if it have not been merged by that time), or start a new PR.